aderyn_core/src/fscloc/engine.rs

```diff
@@ -12,16 +12,17 @@ pub fn count_lines_of_code_and_collect_line_numbers_to_ignore(
 ) -> Mutex<HashMap<String, Stats>> {
     let walker = WalkBuilder::new(src);
     let (tx, rx) = crossbeam_channel::unbounded();
+
     walker.build_parallel().run(|| {
         let tx = tx.clone();
         Box::new(move |res| {
             if let Ok(target) = res {
-                let target_path = target.path().to_str();
                 if target.file_type().unwrap().is_file() {
                     // dbg!(target_path.unwrap());
+                    let remaining = target.path().strip_prefix(&src).unwrap();
                     if src_filepaths
                         .iter()
-                        .any(|fp| target_path.map_or(false, |path| path.contains(fp)))
+                        .any(|fp| &remaining.to_string_lossy().to_string() == fp)
```

Comment:
Corrected overzealous matching which was the reason for non-deterministic cloc

Inside the `docs/` folder you will see a pattern like `src/FjordToken.sol/something.something`  

Which passed for `src/FjordToken.sol/` and was the reason for error.


That is been fixed now

Reply:
Heck, I made it even better. Instead of filtering out the files of interest, we only feed in the files of interest to the WalkBuilder. That makes it a lot faster because we avoid unnecessary scheduling of threads